cmd/bucket-lifecycle.go

```diff
@@ -70,6 +71,29 @@ func NewLifecycleSys() *LifecycleSys {
 	return &LifecycleSys{}
 }
 
+func ilmTrace(startTime time.Time, duration time.Duration, oi ObjectInfo, event string) madmin.TraceInfo {
+	return madmin.TraceInfo{
+		TraceType: madmin.TraceILM,
+		Time:      startTime,
+		NodeName:  globalLocalNodeName,
+		FuncName:  event,
+		Duration:  duration,
+		Path:      fmt.Sprintf("%s?versionId=%s", pathJoin(oi.Bucket, oi.Name), oi.VersionID),
```

Comment:
Per @klauspost recommendation, we can use the `Custom` map to include details like version-id instead of overloading the path, like I had suggested earlier. 
```suggestion
		Path:      pathJoin(oi.Bucket, oi.Name),
		Custom: map[string]string{"version-id": oi.VersionID},
```